crates/ui/src/components/title_bar/linux_window_controls.rs

```diff
@@ -0,0 +1,133 @@
+                r: 0.9,
+                g: 0.9,
+                b: 0.9,
+                a: 0.1,
+            },
+        };
+
+        div()
+            .id("linux-window-controls")
+            .flex()
+            .flex_row()
+            .justify_center()
+            .content_stretch()
+            .max_h(self.button_height)
+            .min_h(self.button_height)
+            .child(TitlebarButton::new(
+                "minimize",
+                TitlebarButtonType::Minimize,
+                button_hover_color,
+            ))
+            .child(TitlebarButton::new(
+                "maximize-or-restore",
+                if cx.is_maximized() {
+                    TitlebarButtonType::Restore
+                } else {
+                    TitlebarButtonType::Maximize
+                },
+                button_hover_color,
+            ))
+            .child(TitlebarButton::new(
+                "close",
+                TitlebarButtonType::Close,
+                close_button_hover_color,
+            ))
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
+enum TitlebarButtonType {
+    Minimize,
+    Restore,
+    Maximize,
+    Close,
+}
+
+#[derive(IntoElement)]
+struct TitlebarButton {
+    id: ElementId,
+    icon: TitlebarButtonType,
+    hover_background_color: Rgba,
+}
+
+impl TitlebarButton {
+    pub fn new(
+        id: impl Into<ElementId>,
+        icon: TitlebarButtonType,
+        hover_background_color: Rgba,
+    ) -> Self {
+        Self {
+            id: id.into(),
+            icon,
+            hover_background_color,
+        }
+    }
+}
+
+impl RenderOnce for TitlebarButton {
+    fn render(self, _cx: &mut WindowContext) -> impl IntoElement {
+        let width = px(36.);
+
+        h_flex()
+            .id(self.id)
+            .justify_center()
+            .content_center()
+            .w(width)
+            .h_full()
+            .hover(|style| style.bg(self.hover_background_color))
+            .active(|style| {
+                let mut active_color = self.hover_background_color;
+                active_color.a *= 0.2;
+
+                style.bg(active_color)
+            })
+            .child(Icon::new(match self.icon {
+                TitlebarButtonType::Minimize => IconName::Dash,
+                TitlebarButtonType::Restore => IconName::Minimize,
+                TitlebarButtonType::Maximize => IconName::Maximize,
+                TitlebarButtonType::Close => IconName::Close,
+            }))
+            .on_click(move |_, cx| {
+                cx.stop_propagation();
+                match self.icon {
+                    TitlebarButtonType::Minimize => cx.minimize_window(),
+                    TitlebarButtonType::Restore => cx.zoom_window(),
+                    TitlebarButtonType::Maximize => cx.zoom_window(),
+                    // TODO: trigger unsaved changes dialog
+                    TitlebarButtonType::Close => cx.remove_window(),
```

Comment:
We need to access the `workspace` crate for that, which is not possible since it depends on this crate :upside_down_face: 

Maybe we'll have to move workspace actions to a separate crate?

Reply:
We can pass down a `Box<dyn Action>` to be dispatched for this behavior, which the workspace can configure when it uses this. 

IDK if this is the ideal solution though, but it's the one I was thinking of.